gallery/src/Application.tsx

```diff
@@ -4,92 +4,116 @@
-import SectionScrollingWrapper from './styles/NxViewportSized/SectionScrollingWrapper';
-
-const pageMappings: PageMapping = mergeAll(values(pageConfig));
-
-function Page({ match, location }: RouteChildrenProps<{ pageName: string }>) {
-  const pageName = match ? match.params.pageName : null,
-      pageHeader = pageName || 'Home',
-      Content = pageName ? pageMappings[pageName] : Home;
-
-  useEffect(function() {
-    handleQueryParams(queryString.parse(location.search));
-  }, [location.search]);
-
-  if (Content) {
-    // Put a key on <main> so that it re-renders entirely on route change, resetting scroll position
-    return (
-      <main key={pageName || 'home'} className="nx-page-main">
-        <div className="nx-page-title">
-          <h1 className="nx-h1">
-            {pageHeader}
-          </h1>
-        </div>
-        <Content/>
-      </main>
-    );
-  }
-  else {
-    // unknown page
-    return <Redirect to="/" />;
-  }
-}
+const logo = require('../src/resources/logos/lifecycle.svg');
 
 function Application() {
-  return (
-    <Router>
-      <PageHeader />
-      <div className="nx-page-content">
-        <Switch>
-          <Route exact path="/PageLevelAlertExample">
-            <NxLoadWrapperPageLevelExample/>
-          </Route>
-          <Route>
-            <aside className="nx-page-sidebar" id="gallery-sidebar">
-              <GalleryNav />
-            </aside>
-            <Switch>
-              <Route path="/pages/:pageName" component={Page} />
-              <Route exact path="/" component={Page} />
+  const [toggleCheck, onToggleCollapse] = useToggle(false);
+  const [toggleCheck2, onToggleCollapse2] = useToggle(false);
 
-              {/* Special cases, these examples need their own page separate from their documentation */}
-              <Route exact path="/NxViewportSizedExample">
-                <SectionScrollingWrapper>
-                  <NxViewportSizedExample />
-                </SectionScrollingWrapper>
-              </Route>
-              <Route exact path="/NxViewportSizedExpandingExample">
-                <SectionScrollingWrapper>
-                  <NxViewportSizedExpandingExample />
-                </SectionScrollingWrapper>
-              </Route>
-              <Redirect to="/" />
-            </Switch>
-          </Route>
-        </Switch>
+  return (
+    <div id="temporary-navigation-sidebar-example" className="nx-page-content nx-page-content--full-width">
+      <NxStatefulNavigationSidebar isDefaultOpen={true}
+                                   toggleOpenIcon={faArrowLeft}
+                                   toggleCloseIcon={faCrow}
+                                   logoAltText="product name"
+                                   logoLink="/home"
+                                   logoImg={logo}>
+        <NxNavigationSidebarLinks>
+          <a href="#alpha" className="nx-sidebar-navigation__link selected nx-text-link">
```

Comment:
It would be possible to get rid of `.nx-sidebar-navigation__link` entirely in favor of `.nx-sidebar-navigation .nx-text-link` descendant selectors.  Not sure if that would be considered better though from a BEM perspective.  @glenhunter what do you think?

Reply:
> It would be possible to get rid of `.nx-sidebar-navigation__link` entirely in favor of `.nx-sidebar-navigation .nx-text-link` descendant selectors. Not sure if that would be considered better though from a BEM perspective. @glenhunter what do you think?

It's not really a BEM question but more of a specificity question. The advantage of `.nx-sidebar-navigation__link` is that it doesn't have to be nested and is easier to override if necessary. 

FWIW I wouldn't be opposed to `nx-text-link--sidebar-nav`, which also wouldn't require nesting.

Reply:
> > It would be possible to get rid of `.nx-sidebar-navigation__link` entirely in favor of `.nx-sidebar-navigation .nx-text-link` descendant selectors. Not sure if that would be considered better though from a BEM perspective. @glenhunter what do you think?
> 
> It's not really a BEM question but more of a specificity question. The advantage of `.nx-sidebar-navigation__link` is that it doesn't have to be nested and is easier to override if necessary.
> 
> FWIW I wouldn't be opposed to `nx-text-link--sidebar-nav`, which also wouldn't require nesting.

Why do we _need_ to get rid of `nx-sidebar-navigation__link`? It's a class that's being used, whose name is descriptive enough of its intention IMO, and also follows the naming convention.

Reply:
> Why do we need to get rid of nx-sidebar-navigation__link? It's a class that's being used, whose name is descriptive enough of its intention IMO, and also follows the naming convention.

we don't necessarily I was just worried about the complexity here.  Now that that complexity is encapsulated within a component I'd say forget about it.